drivers/flash/flash_stm32_qspi.c

```diff
@@ -1251,8 +1327,8 @@ static int qspi_program_quad_io(const struct device *dev)
 		return ret;
 	}
 
-	if ((reg & qe_bit) == 0U) {
-		LOG_ERR("Status Register %u [0x%02x] not set", qe_reg_num, reg);
+	if (!flash_reg_is_set_for_all(&reg, qe_bit)) {
+		LOG_ERR("Status Register %u [0x%02x] not set", qe_reg_num, flash_reg_to_raw(&reg));
```

Comment:
The status register log output with format `%02x` will crop `uint16_t` to `uint8_t` (_for `STM32_QSPI_DOUBLE_FLASH`_); is that desired?